homeassistant/components/zha/config_flow.py

```diff
@@ -735,10 +788,11 @@ async def async_step_restore_backup(
             # User unplugged the new adapter, allow retry
             return self.async_show_progress_done(next_step_id="pre_plug_in_new_radio")
         except CannotWriteNetworkSettings as exc:
-            return self.async_abort(
+            self._progress_error = AbortFlow(
```

Comment:
If this didn't fail any tests before, it means we're lacking tests for this case.

Reply:
Yep, kind of. The test for this currently doesn't raise the "invalid transition" ValueError, because the progress flow never really starts, as we instantly raise `CannotWriteNetworkSettings` in the test. 

Just adding `asyncio.sleep(0)` to the `side_effect` before raising `CannotWriteNetworkSettings` causes the test to fail as expected, without this PR.
So, we should add something along those lines to the test to make sure the progress flow has a chance to start.

For reference, the test for this is:
https://github.com/home-assistant/core/blob/c0db966afd1b5b8ddf3689ad9dec962cfe5ef525/tests/components/zha/test_config_flow.py#L721-L739